### ISSUE TITLE
Run operations in parallel when possible

### ISSUE THREAD

**Issue by ellismg** [Azure/azure-dev#372]
During the code review of the monitor spike we started to discuss cases where we have to do multiple azure operations which could happen in parallel (for example, listing resources in an application that contains multiple resource groups means we'll have to call a single list call per resource group) but are done today serially.

We decided to defer this work for now, since in the common case we don't have a bunch of resource groups and hence the ability to exploit parallelism but this would be something that should be helpful which improves performance.  This issue tracks that work.

See https://github.com/Azure/azure-dev/pull/391#pullrequestreview-852180297 for more discussion.

**Comment by spboyer** [comment]
partially addressed by #6200

**Comment by spboyer** [comment]
@vhvb1989 - add to concurx issue

**Comment by jongio** [comment]
Closing as superseded by epic #7659 (azd performance umbrella).

This 2022 issue was a generic catchall; the specific work it described is now tracked under #7659 with concrete sub-issues:
- Parallel resource group listing -> #7659 scope
- Parallel infra layer provisioning -> #6291
- Parallel service deployment -> addressed by recent in-flight PRs (see #7659)

No scope is lost; only tracking duplication is removed.
